src/Headless/Avalonia.Headless.NUnit/AvaloniaTestMethodCommand.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
+using Avalonia.Threading;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Commands;
+
+namespace Avalonia.Headless.NUnit;
+
+internal class AvaloniaTestMethodCommand : DelegatingTestCommand
+{
+    private readonly HeadlessUnitTestSession _session;
+    private readonly List<Action> _beforeTest;
+    private readonly List<Action> _afterTest;
+
+    private static FieldInfo s_innerCommand = typeof(DelegatingTestCommand)
+        .GetField("innerCommand", BindingFlags.Instance | BindingFlags.NonPublic)!;
```

Comment:
This is gnarly, but @maxkatz6 doesn't think there's currently a better way to do it with nunit.

Reply:
Yes, will add a comment with details and links.